common/hello.go

```diff
@@ -0,0 +1,190 @@
+				if err != nil {
+					return err
+				}
+				return checkAppStatus(app)
+			}).
+				WithTimeout(3 * time.Minute).
+				WithPolling(5 * time.Second).
+				Should(Succeed())
+
+			Eventually(func() error {
+				app, err := state.GetFramework().GetApp(ctx, fmt.Sprintf("%s-external-dns", state.GetCluster().Name), org.GetNamespace())
+				if err != nil {
+					return err
+				}
+				return checkAppStatus(app)
+			}).
+				WithTimeout(3 * time.Minute).
+				WithPolling(5 * time.Second).
+				Should(Succeed())
+
+			// `external-dns` will only create dns records for Ingresses on the `kube-system` namespace, that's why we install the nginx app in that namespace.
+			nginxApp, nginxConfigMap = deployApp(ctx, "ingress-nginx", "kube-system", org, "3.0.0", "", map[string]string{})
+			Eventually(func() error {
+				app, err := state.GetFramework().GetApp(ctx, nginxApp.Name, nginxApp.Namespace)
+				if err != nil {
+					return err
+				}
+				return checkAppStatus(app)
+			}).
+				WithTimeout(3 * time.Minute).
+				WithPolling(5 * time.Second).
+				Should(Succeed())
+
+			helloWorldIngressHost = fmt.Sprintf("hello-world.%s", getWorkloadClusterDnsZone(ctx))
+			helloAppValues := map[string]string{"IngressUrl": helloWorldIngressHost}
+			helloApp, helloConfigMap = deployApp(ctx, "hello-world", "giantswarm", org, "2.0.0", "./test_data/helloworld_values.yaml", helloAppValues)
+			Eventually(func() error {
+				// The hello-world app creates Ingress resources, and they won't be able to be created until the nginx webhooks are up and running, so the first time we try to install the hello-world app, it will fail.
+				// Meaning that it won't be retried until the next chart-operator reconciliation loop 5 minutes later.
+				// To speed things up, we keep patching the hello-world `App` CR by adding a label. That way, we trigger reconciliation loops in chart-operator.
+				now := time.Now()
+				patchedApp := helloApp.DeepCopy()
+				labels := patchedApp.GetLabels()
+				labels["update"] = fmt.Sprintf("%d", now.Unix())
+				patchedApp.SetLabels(labels)
+				managementClusterKubeClient := state.GetFramework().MC()
+				err := managementClusterKubeClient.Patch(ctx, patchedApp, ctrl.MergeFrom(helloApp))
+				if err != nil {
+					return err
+				}
+				app, err := state.GetFramework().GetApp(ctx, helloApp.Name, helloApp.Namespace)
+				if err != nil {
+					return err
+				}
+				return checkAppStatus(app)
+			}).
+				WithTimeout(6 * time.Minute).
+				WithPolling(5 * time.Second).
+				Should(Succeed())
+		})
+
+		It("hello world app responds successfully", func() {
+			if !externalDnsSupported {
+				Skip("Bastion is not supported.")
+			}
+
+			Eventually(func() (*http.Response, error) {
+				helloWorldIngressUrl := fmt.Sprintf("https://%s", helloWorldIngressHost)
+				logger.Log("Trying to get a successful response from %s", helloWorldIngressUrl)
+				return http.Get(helloWorldIngressUrl)
+			}, "10m", "5s").Should(HaveHTTPStatus(http.StatusOK))
```

Comment:
There's the nginx-ingress _default backend_ which might also respond with HTTP 200, but that would mean a problem since the request didn't reach the Hello World app. Let's please check the content of the response to contain a certain string. [This](https://github.com/giantswarm/helloworld/blob/master/content/index.html) seems to be what the app would return.

Reply:
Maybe @mcharriere can confirm, but I was under the impression that the default backend will return 404.

Reply:
yes, I can confirm, `default-backend` always returns 404.

Reply:
I'd still suggest you check the actual content. Only then, you don't have any assumptions but know for sure you're testing the right thing.

Reply:
Done